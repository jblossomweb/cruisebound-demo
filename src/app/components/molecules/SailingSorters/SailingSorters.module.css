.sorters {
}

.sorters label {
  @apply inline
    mr-2;
}
