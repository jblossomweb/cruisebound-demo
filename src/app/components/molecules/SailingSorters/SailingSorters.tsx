import React from 'react'

import type SortValue from 'app/types/SortValue'
import Dropdown from 'app/components/atoms/Dropdown'
import styles from './SailingSorters.module.css'

export interface SailingSortersProps {
  sortValue?: SortValue
  // eslint-disable-next-line no-unused-vars
  onSort: (value: SortValue) => void
}

const SailingSorters: React.FC<SailingSortersProps> = ({
  sortValue = undefined,
  onSort,
}) => (
  <div className={styles.sorters}>
    <Dropdown
      label="Sort by"
      value={sortValue}
      options={[
        {
          value: 'rating',
          display: 'Rating (highest first)',
        },
        {
          value: 'rating-asc',
          display: 'Rating (lowest first)',
        },
        {
          value: 'price',
          display: 'Price (lowest first)',
        },
        {
          value: 'price-desc',
          display: 'Price (highest first)',
        },
        {
          value: 'duration',
          display: 'Duration (longest first)',
        },
        {
          value: 'duration-asc',
          display: 'Duration (shortest first)',
        },
        {
          value: 'departure',
          display: 'Departure Date (earliest first)',
        },
        {
          value: 'departure-desc',
          display: 'Departure Date (latest first)',
        },
      ]}
      onSelect={(value: string) => {
        onSort(value as SortValue)
      }}
      shadow
    />
  </div>
)

export default SailingSorters
