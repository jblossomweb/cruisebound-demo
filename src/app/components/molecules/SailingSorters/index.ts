export type { SailingSortersProps, SortValue } from './SailingSorters'
export { default } from './SailingSorters'
