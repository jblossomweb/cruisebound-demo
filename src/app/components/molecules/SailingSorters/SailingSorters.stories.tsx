import React from 'react'
import type { StoryFn, Meta } from '@storybook/react'

import SailingSorters, { type SailingSortersProps } from './SailingSorters'

export default {
  title: 'components/molecules/SailingSorters',
  component: SailingSorters,
  argTypes: {
    sortValue: { control: 'text' },
    onSort: { control: 'none' },
  },
} as Meta<typeof SailingSorters>

const Template: StoryFn<typeof SailingSorters> = (
  args: SailingSortersProps
) => <SailingSorters {...args} />

export const Sample = Template.bind({})
Sample.args = {
  onSort: () => {
    // void
  },
}

export const Controlled = Template.bind({})
Controlled.args = {
  ...Sample.args,
  sortValue: 'price',
}
