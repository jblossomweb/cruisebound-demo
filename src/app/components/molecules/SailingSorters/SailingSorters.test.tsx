/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import storySnaps from 'test/utils/storySnaps'

import SailingSorters from './SailingSorters'
import * as stories from './SailingSorters.stories'

describe('app/components/molecules/SailingSorters', () => {
  describe('stories', () => {
    storySnaps(stories)
  })

  describe('interaction', () => {
    afterEach(cleanup)

    it('should call onSort when selection changes', () => {
      const onSort = jest.fn()
      expect(onSort).not.toHaveBeenCalled()
      render(<SailingSorters onSort={onSort} />)
      expect(onSort).not.toHaveBeenCalled()
      const select = screen.getByTestId('select')
      fireEvent.change(select, { target: { value: 'price' } })
      expect(onSort).toHaveBeenCalled()
      expect(onSort).toHaveBeenCalledWith('price')
    })
  })
})
