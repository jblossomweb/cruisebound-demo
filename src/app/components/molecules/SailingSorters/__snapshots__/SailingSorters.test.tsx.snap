// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app/components/molecules/SailingSorters stories Controlled matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="sorters"
  >
    <label
      class="label"
    >
      Sort by
    </label>
    <div
      class="select shadow"
    >
      <select
        class="inner"
        data-testid="select"
      >
        <option
          value="rating"
        >
          Rating (highest first)
        </option>
        <option
          value="price"
        >
          Price (lowest first)
        </option>
        <option
          value="duration"
        >
          Duration (longest first)
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`app/components/molecules/SailingSorters stories Sample matches markup snapshot 1`] = `
<DocumentFragment>
  <div
    class="sorters"
  >
    <label
      class="label"
    >
      Sort by
    </label>
    <div
      class="select shadow"
    >
      <select
        class="inner"
        data-testid="select"
      >
        <option
          value="rating"
        >
          Rating (highest first)
        </option>
        <option
          value="price"
        >
          Price (lowest first)
        </option>
        <option
          value="duration"
        >
          Duration (longest first)
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
